plane-detection.bs

```diff
@@ -115,22 +127,208 @@ Feature descriptor {#anchor-feature-descriptor}
 
 In order for the applications to signal their interest in using plane detection during a session, the session must be requested with appropriate [=feature descriptor=]. The string <dfn>plane-detection</dfn> is introduced by this module as a new valid feature descriptor for plane detection feature.
 
-A device is [=capable of supporting=] the plane-detection feature if the device's tracking system exposes a [[#native-plane-detection]] capability. The [=inline XR device=] MUST NOT be treated as [=capable of supporting=] the plane-detection feature.
+A device is [=capable of supporting=] the plane-detection feature if the device's tracking system exposes a [=native plane detection=] capability. The [=inline XR device=] MUST NOT be treated as [=capable of supporting=] the plane-detection feature.
+
+When a session is created with plane-detection feature enabled, the [=update planes=] algorithm MUST be added to the [=list of frame updates=] of that session.
+
+<div class="example">
+The following code demonstrates how a session that requires plane detection could be requested:
+
+<pre highlight="js">
+const session = await navigator.xr.requestSession("immersive-ar", {
+  requiredFeatures: ["plane-detection"]
+});
+</pre>
+
+</div>
+
+Planes {#planes-section}
+======
+
+XRPlaneOrientation {#plane-orientation}
+------------------
+
+<script type="idl">
+enum XRPlaneOrientation {
+    "horizontal",
+    "vertical"
+};
+</script>
+
+- A plane orientation of <dfn enum-value for="XRPlaneOrientation">"horizontal"</dfn> indicates that the plane is primarily oriented horizontally (according to the conventions of the underlying platform).
+- A plane orientation of <dfn enum-value for="XRPlaneOrientation">"vertical"</dfn> indicates that the plane is primarily oriented vertically (according to the conventions of the underlying platform).
+
+XRPlane {#plane}
+-------
+
+<script type="idl">
+interface XRPlane {
+    [SameObject] readonly attribute XRSpace planeSpace;
+
+    readonly attribute FrozenArray<DOMPointReadOnly> polygon;
+    readonly attribute XRPlaneOrientation? orientation;
+    readonly attribute DOMHighResTimeStamp lastChangedTime;
```

Comment:
Instead of basing logic on `lastChangedTime`, would it be easier for the author if they got lists of new, updated and deleted planes?

Reply:
Difference lists (and events) were mentioned in #4. I think we should initially go with `lastChangedTime` that can be used to polyfill both difference lists and events - this way, we'll provide an MVP that we could still add to, based on developer feedback (I'd like to send an Origin Trial for this to hopefully learn what people are missing from the API). WDYT?

Reply:
Wow, your [explainer](https://github.com/immersive-web/real-world-geometry/blob/master/planes-notifying-about-removal.md) really goes in depth over this option :-)
It seems you were leaning towards the attribute-based approach in that spec. Maybe you can update it with a rationale why you switched to a timestamp?

Reply:
Will do, I mainly wanted to avoid event-based approach because it seemed more complex than it has to be, and the timestamp approach allows us to postpone the decision until we can get some feedback about what the API should look like from the developers.